### ISSUE TITLE
Back button in the podcast search does not have an accessibility label

### ISSUE THREAD

**Issue by MiSikora** [Automattic/pocket-casts-android#1711]
### Description

When searching for podcasts and using TalkBack function its hard to navigate around the search function because the back button is unlabelled.

This button is missing content description.

https://github.com/Automattic/pocket-casts-android/blob/703fa701b435e35e42d04367f33cdc391b1963be/modules/features/search/src/main/res/layout/fragment_search.xml#L42-L50

### Step-by-step reproduction instructions

1. Enable TalkBack in the device settings.
2. Open the app and navigate to the podcast search feature.
3. Tap on the back button.
4. You'll hear "Unlabeled".

### Screenshots or screen recording

I added a recording but Android doesn't capture audio from TalkBack, unfortunately. However, it is super easy to reproduce.

https://github.com/Automattic/pocket-casts-android/assets/30936061/748f865b-c779-4915-8bee-26bbfe6ccae0

### Did you search for existing bug reports?

- [X] I have searched for existing bug reports.

### Device, Operating system, and Pocket Casts app version

Google Pixel 6, Android 13, Pocket Casts `7.54 (9180)`
